{{ cookiecutter.__project_name_kebab }}/.pre-commit-config.yaml

```
@@ -18,30 +18,13 @@ repos:
         args: ['--unsafe']
       - id: end-of-file-fixer
       - id: trailing-whitespace
-  - repo: https://github.com/psf/black
-    rev: 23.9.1
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    # ruff config is in ruff.toml
+    rev: v0.1.14
     hooks:
-      - id: black
-        language_version: python3
-        args: ['--target-version', 'py38']
-  - repo: https://github.com/adamchainz/blacken-docs
-    rev: 1.16.0
-    hooks:
-      - id: blacken-docs
-        additional_dependencies: [black==23.9.1]
-  - repo: https://github.com/pycqa/isort
-    # isort config is in setup.cfg
-    rev: 5.12.0
-    hooks:
-      - id: isort
-  - repo: https://github.com/pycqa/flake8
-    # flake8 config is in setup.cfg
-    rev: 6.1.0
-    hooks:
-      - id: flake8
-        additional_dependencies:
-          - flake8-bugbear
-          - flake8-comprehensions
+      - id: ruff
+        args: [--fix]
```

what about:

```suggestion
        args: [--fix, --exit-non-zero-on-fix]
```

?

---

Interesting. I didn't do this because the suggestion in https://github.com/astral-sh/ruff-pre-commit?tab=readme-ov-file#using-ruff-with-pre-commit is just `--fix`. I have no idea what a pre-commit enthusiast would expect to happen here.

---

I guess this is my personal preference in that I like to see the changes before they are committed. Let's leave as is